lib/maths/common/CQuantileSketch.cc

```diff
@@ -99,19 +177,34 @@ std::ptrdiff_t nextDifferent(TFloatFloatPrVec& knots, std::size_t index) {
     return next.index();
 }
 
+std::size_t fastSketchSize(double reductionFactor, std::size_t size) {
+    size = static_cast<std::size_t>(
+        static_cast<double>(size) * CQuantileSketch::REDUCTION_FACTOR / reductionFactor + 0.5);
+    return size + (3 - (size + 1) % 3) % 3;
```

Comment:
Where does `3` come from? Can we use a constant here?

Reply:
This ensures that the loop which computes the weights never have any left over costs to compute, since we take steps of size 3.